### PR TITLE
Cleanup unused code

### PR DESCRIPTION
**Pull request by trbehera**
Remove unused code that are no longer needed as per
new implementation.

This includes legacy code used in SDO implementation as 
well as the code for unused features.

Signed-off-by: Behera, Tushar Ranjan <tushar.ranjan.behera@intel.com>

### CODE REVIEW

File: lib/include/fdoprot.h [hunk 1]
```diff
@@ -109,6 +96,7 @@
 #define FDO_TYPE_ERROR 255
 
 // Persistent
+// TODO: Needs review (Only FDO_TYPE_HMAC is used in code)
```
Comment (johneasterday) [inline lib/include/fdoprot.h:99]: should this comment be removed? What is the TODO?

File: lib/include/fdoprot.h [hunk 2]
```diff
@@ -117,19 +105,13 @@
 #define FDO_TYPE_DEVICE_CRED 6
 #define FDO_TYPE_HMAC 7
 
-// Report
-#define FDO_MGR_AGENT_SND_REPORT 52
-#define FDO_MGR_AGENT_RCV_REPORT_ACK 53
-
 // For restful URL mapping
+// TODO: Not used, needs to review
```
Comment (johneasterday) [inline lib/include/fdoprot.h:109]: should this comment be removed? What is the TODO?

File: lib/include/fdoprot.h [hunk 3]
```diff
@@ -117,19 +105,13 @@
...
 #define FDOMsg_typeMIN FDO_TO1_TYPE_HELLO_FDO
 #define FDOMsg_typeMAX FDO_TO2_DONE2
 
-// Protocol version
-#define FDO_VER_MAJOR 1
-#define FDO_VER_MINOR 10
-
 // Error Message
+// TODO: Not used, needs to review (Only INTERNAL_SERVER_ERROR is used)
```
Comment (johneasterday) [inline lib/include/fdoprot.h:114]: should this comment be removed? What is the TODO?

File: lib/include/fdoprot.h [hunk 4]
```diff
@@ -155,6 +137,7 @@
  *
  * An alternative is to declare the buffer global
  */
+// TODO: Needs review, macro not used.
```
Comment (johneasterday) [inline lib/include/fdoprot.h:140]: should this comment be removed? What is the TODO?

File: tests/unit/test_cryptoSupport.c [hunk 5]
```diff
@@ -17,7 +17,7 @@
 #include "ecdsa_privkey.h"
 #include "safe_lib.h"
 #include "fdotypes.h"
-#include "test_RSARoutines.h"
+//#include "test_RSARoutines.h"
```
Comment (johneasterday) [inline tests/unit/test_cryptoSupport.c:20]: shouldn't this line be deleted?

File: cmake/blob_path.cmake [hunk 6]
```diff
@@ -16,7 +16,6 @@ if(TARGET_OS MATCHES linux)
     -DPLATFORM_IV=\"${BLOB_PATH}/data/platform_iv.bin\"
     -DPLATFORM_HMAC_KEY=\"${BLOB_PATH}/data/platform_hmac_key.bin\"
     -DPLATFORM_AES_KEY=\"${BLOB_PATH}/data/platform_aes_key.bin\"
-    -DEPID_PRIVKEY=\"${BLOB_PATH}/data/epidprivkey.dat\"
     -DFDO_CRED=\"${BLOB_PATH}/data/PMDeviceCredentials.bin\"
```
Comment (prateekcr) [inline cmake/blob_path.cmake:19]: Seems like FDO_CRED isn't used anywhere either. Please remove this as well, here and Line 105.

File: lib/include/fdoprot.h (hunk 1 above)
Comment (prateekcr) [reply lib/include/fdoprot.h:99]: These can be removed, including the used 'FDO_TYPE_HMAC' and its usage. They don't really serve any purpose currently (unused in older Client SDK as well).

File: lib/include/fdoprot.h (hunk 4 above)
Comment (prateekcr) [reply lib/include/fdoprot.h:140]: This was earlier used to hold content that could be printed on console (nonce, GUID etc.). We've removed those print statements altogether, hence this variable is no longer used. Can be removed.

File: lib/include/fdoprot.h (hunk 1 above)
Comment (trbehera) [reply lib/include/fdoprot.h:99]: Fixed in commit 1c24ccaba0e54c28b18fdfaa30a87b1cad086e07

File: lib/include/fdoprot.h (hunk 2 above)
Comment (trbehera) [reply lib/include/fdoprot.h:109]: Updated in commit 1c24ccaba0e54c28b18fdfaa30a87b1cad086e07

File: lib/include/fdoprot.h (hunk 3 above)
Comment (trbehera) [reply lib/include/fdoprot.h:114]: Updated the TODO statement that it should be mapped to the error messages.

Reflected in commit 1c24ccaba0e54c28b18fdfaa30a87b1cad086e07

File: lib/include/fdoprot.h (hunk 4 above)
Comment (trbehera) [reply lib/include/fdoprot.h:140]: Removed the macro and related comment.

Updated in commit 1c24ccaba0e54c28b18fdfaa30a87b1cad086e07

File: tests/unit/test_cryptoSupport.c (hunk 5 above)
Comment (trbehera) [reply tests/unit/test_cryptoSupport.c:20]: Updated in commit 32d6b292386138a719c01fb05eb81ef16ec7d2f6

File: cmake/blob_path.cmake (hunk 6 above)
Comment (trbehera) [reply cmake/blob_path.cmake:19]: Updated in 4d168a5817db6fc30224708a431dee89d01fe254

**Review by prateekcr — APPROVED** [review]

**Review by johneasterday — APPROVED** [review]

**Pull request merged**